server/src/daemon/item.rs

```diff
@@ -54,16 +54,20 @@ impl Item {
     pub async fn secret(&self, session: ObjectPath<'_>) -> Result<SecretInner> {
         let inner = self.inner.read().await;
         let secret = inner.secret();
-        let session = self.manager.lock().unwrap().session(session).unwrap();
         let parameters = self.parameters();
         let content_type = self.content_type();
-
-        Ok(SecretInner(
-            session.path().into(),
-            parameters.to_vec(),
-            secret.to_vec(),
-            content_type.to_owned(),
-        ))
+        match self.manager.lock().unwrap().session(session.clone()) {
+            Some(session) => Ok(SecretInner(
+                session.path().into(),
+                parameters.to_vec(),
+                secret.to_vec(),
+                content_type.to_owned(),
+            )),
+            None => {
+                tracing::error!("Session {session} not found");
+                Err(ServiceError::NoSuchObject)
```

Comment:
This should be NoSession instead. The NoSuchObject is only used for items/collections. See the docs.

Reply:
Got it, thanks!